language/movie_nl/main.py

```diff
@@ -0,0 +1,345 @@
+    else:
+        pol = sentiment.get("polarity")
+        mag = sentiment.get("magnitude")
+
+    if pol is None and mag is not None:
+        pol = 0
+    return (pol, mag)
+
+
+class Document(object):
+    """Document class captures a single document of movie reviews."""
+
+    def __init__(self, text, doc_id, doc_path):
+        self.text = text
+        self.doc_id = doc_id
+        self.doc_path = doc_path
+        self.sentent_pair = None
+        self.label = None
+
+    def extract_all_sentences(self, service):
+        """Extract the sentences in a document."""
+
+        if self.sentent_pair is None:
+            docs = service.documents()
+            request_body = get_request_body(
+                self.text,
+                syntax=True,
+                entities=True,
+                sentiment=False)
+            request = docs.annotateText(body=request_body)
+
+            ent_list = []
+
+            response = request.execute()
+            entities = response.get('entities', [])
+            sentences = response.get('sentences', [])
+
+            sent_list = [
+                sentence.get('text').get('content')for sentence in sentences
+            ]
+
+        for entity in entities:
+            ent_type = entity.get('type')
+            wiki_url = entity.get('metadata', {}).get('wikipedia_url')
+
+            if ent_type == 'PERSON' and wiki_url is not None:
+                ent_list.append(wiki_url)
+
+            self.sentent_pair = (sent_list, ent_list)
+
+        return self.sentent_pair
+
+
+def to_sentiment_json(doc_id, sent, label):
+    """Convert the sentiment info to json."""
+    json_doc = {}
+
+    json_doc['doc_id'] = doc_id
+    json_doc['sentiment'] = float('%.3f' % sent)
+    json_doc['label'] = label
+
+    return json.dumps(json_doc)
+
+
+def get_wiki_title(wiki_url):
+    """Get the wikipedia page title for a given wikipedia URL."""
+    try:
+        content = requests.get(wiki_url).text
+        return content.split('title')[1].split('-')[0].split('>')[1].strip()
+    except:
+        return os.path.basename(wiki_url).replace('_', ' ')
+
+
+def to_entity_json(entity, e_tuple):
+    """Convert the entity info to json."""
+    json_doc = {}
+
+    avg_sentiment = float(e_tuple[0]) / float(e_tuple[1])
+
+    json_doc['wiki_url'] = entity
+    json_doc['name'] = get_wiki_title(entity)
+    json_doc['sentiment'] = float('%.3f' % e_tuple[0])
+    json_doc['avg_sentiment'] = float('%.3f' % avg_sentiment)
+
+    return json.dumps(json_doc)
+
+
+def get_sentiment_entities(service, document):
+    """Compute the overall sentiment volume in the document"""
+    sentiments, entities = analyze_document(service, document)
+
+    sentiments = [sent for sent in sentiments if sent[0] is not None]
+    negative_sentiments = [
+        sentiment for sentiment in sentiments if sentiment[0] < 0.0]
+    positive_sentiments = [
+        sentiment for sentiment in sentiments if sentiment[0] >= 0.0]
```

Comment:
Well, it's not negative? ;)
Probably doesn't matter either way, since it's summed up later.
Actually, broke out the tuple to make things clear, and simplified the sum later on. PTAL